app/routes/posts/form.jsx

```diff
@@ -0,0 +1,85 @@
+import { redirect } from "@remix-run/node"
+import {useState} from 'react'
+
+export const action = async ({request}) => {
+  const form = await request.formData();
+  const name = form.get('first-name')
+  const email = form.get('email')
+  const inPerson = form.get('in-person')
+  const online = form.get('online')
+  const pizza = form.get('pizza')
+
+  const fields = {name, email, inPerson, online, pizza}
+
+  const post = await db.post.create({data: fields})
+  
+  return redirect('/')
+}
+
+export default function Forms(){
+
+  const [isChecked, setIsChecked] = useState(false)
+
+
+function handleClick(e){
+  if(e.target.value === 'radio1'){
+    setIsChecked(true)
+  } else {
+    document.getElementById('pizza').checked = false
```

Comment:
This might cause an issue with components getting unmounted and remounted. Because this getter being the direct DOM reference the property `checked` will only be sent in the render where the handleClick was called. But if the component were to unmount and remount, say on another page the reference might be lost.

This is probably a good use case for a `useState`. The form will still searlize it because it's a named input. But you can still have component state that ties up the desired UX of the in-person/virtual event.

```jsx
const [pizzaEnabled, setPizzaEnabled] = React.useState(true)

function handleChangeAttendance(evt) {
  setPizzaEnabled(evt.target.value === 'inperson')
}
 ```

Then in your radio group inputs you use the `onChange` to call `handleChangeAttendance`, and the checkbox element you only need to add the prop `defaultChecked={true}`. This will check the pizza checkbox on mount, changing the attendance will only change the disabled state of the checkbox. I don't actually care about the answer if they select virtual. So if it's checked or unchecked it will remain in that state when the attendance changes.

This means the only state that were tracking is which attendance option is selected to only control the state of the food, not the value. The form will still to its thing and we get a nice UX.